Realm/RLMObjectBase.mm

```diff
@@ -441,3 +444,30 @@ + (NSArray *)requiredPropertiesForClass:(Class)cls {
 }
 
 @end
+
+@interface RLMObjectBase (Handover) <RLMThreadConfined_Private>
+@end
+
+@implementation RLMObjectBase (Handover)
+
+- (realm::AnyThreadConfined)rlm_handoverData {
+    // FIXME: Once merged, should be easier to get schema ref (https://github.com/realm/realm-cocoa/pull/3941)
+    auto className = RLMStringDataWithNSString(self.objectSchema.className);
+    auto& schema = _realm->_realm->schema();
+    auto objectSchema = schema.find(std::move(className));
+    REALM_ASSERT(objectSchema != schema.end());
+    return AnyThreadConfined(Object(_realm->_realm, *objectSchema, _row));
+}
+
+- (NSNull *)rlm_handoverMetadata {
+    return [NSNull null];
```

Comment:
`ram_handoverMetadata` only has a non-nullable return type because you chose to mark it as such. IMO it's clearer to mark the result as nullable and return `nil` than it is to claim it's non-nullable and just use a different value for null. `NSNull` is typically only used in places where `nil` is not valid.